helix-view/src/editor.rs

```diff
@@ -34,6 +34,46 @@ where
     Ok(Duration::from_millis(millis))
 }
 
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+#[serde(rename_all = "kebab-case", default, deny_unknown_fields)]
+pub struct FilePickerConfig {
+    /// IgnoreOptions
+    /// Enables ignoring hidden files.
+    /// Whether to hide, that is, to stop hidden files from displaying in file picker. Defaults to false.
+    pub hidden: bool,
+    /// Enables reading ignore files from parent directories.
+    pub parents: bool,
+    /// Enables reading `.ignore` files.
+    /// Whether to hide files listed in .ignore from displaying in file picker. Defaults to false.
+    pub ignore: bool,
+    /// Enables reading `.gitignore` files.
+    /// Whether to hide files in .gitignore from displaying in file picker. Defaults to false.
+    pub git_ignore: bool,
+    /// Enables reading global .gitignore, whose path is specified in git's config: `core.excludefile` option.
+    /// Whether to hide files in global .gitignore from displaying in file picker. Defaults to false.
+    pub git_global: bool,
+    /// Enables reading `.git/info/exclude` files.
+    /// Whether to hide files in .git/info/exclude from displaying in file picker. Defaults to false.
+    pub git_exclude: bool,
+    /// WalkBuilder options
+    /// Maximum Depth to recurse.
+    pub max_depth: Option<usize>,
+}
+
+impl Default for FilePickerConfig {
+    fn default() -> Self {
+        Self {
+            hidden: true,
+            parents: true,
+            ignore: true,
+            git_ignore: true,
+            git_global: true,
+            git_exclude: true,
+            max_depth: None,
+        }
+    }
+}
```

Comment:
I made some updates to these comments and to the book, hope they are more clear and accurate!

Reply:
I think we should get rid of mentioning defaults in comments altogether since it could get outdated, and for the book we could do something like #1119.